WooCommerce/WooCommerceTests/ViewRelated/Search/Order/OrderSearchStarterViewModelTests.swift

```diff
@@ -0,0 +1,98 @@
+
+import XCTest
+import Foundation
+import Yosemite
+import Storage
+
+@testable import WooCommerce
+
+/// Tests for `OrderSearchStarterViewModel`
+///
+final class OrderSearchStarterViewModelTests: XCTestCase {
+    private let siteID: Int64 = 1_000_000
+
+    private var storageManager: StorageManagerType!
+
+    private var storage: StorageType {
+        storageManager.viewStorage
+    }
+
+    override func setUp() {
+        storageManager = MockupStorageManager()
+        super.setUp()
+    }
+
+    override func tearDown() {
+        storageManager = nil
+        super.tearDown()
+    }
+
+    func testItLoadsAllTheOrderStatusForTheGivenSite() {
+        // Given
+        let viewModel = OrderSearchStarterViewModel(siteID: siteID, storageManager: storageManager)
+
+        let expectedItems = [
+            insertOrderStatus(siteID: siteID, status: .completed),
+            insertOrderStatus(siteID: siteID, status: .failed),
+            insertOrderStatus(siteID: siteID, status: .processing),
+        ]
+
+        let unexpectedItem = insertOrderStatus(siteID: 511_315, status: .pending)
+
+        // When
+        viewModel.activateAndForwardUpdates(to: UITableView())
+
+        // Then
+        XCTAssertEqual(viewModel.numberOfObjects, expectedItems.count)
+        XCTAssertFalse(viewModel.fetchedOrderStatuses.contains(where: { $0.siteID != siteID }))
```

Comment:
nit: I was wondering if we could directly compare `viewModel.fetchedOrderStatuses` with `expectedItems`?

Reply:
I like that. Added in 8a95e68f58d89544a3f3723ed8fe1d3a732654a7. I'm keeping the `$0.siteID != siteID` check since it's sort of another proof that all fetched objects are from the given `siteID`. It also protects us if I make a mistake in the `expectedItems` setup.